File: Problems/No.2383_Minimum_Hours_of_Training_to_Win_a_Competition/minimum_hours_of_training_to_win_a_competition.py
class Solution(object):
    def minNumberOfHours(self, initialEnergy, initialExperience, energy, experience):
        """
        :type initialEnergy: int
        :type initialExperience: int
        :type energy: List[int]
        :type experience: List[int]
        :rtype: int
        """

        # parameters
        ExtraEnergy = (-1)*initialEnergy
        ExtraExp, ExpGap = 0, 0

        # start computing
        for id in range(len(energy)):
            # compute energy
            ExtraEnergy += energy[id]
            # compute experience
            if (experience[id] >= initialExperience):
                ExpGap = experience[id] - initialExperience + 1
                ExtraExp += ExpGap
                initialExperience += experience[id] + ExpGap
            else:
                initialExperience += experience[id]
        
        return (max((ExtraEnergy + 1), 0) + ExtraExp)

File: Problems/No.2383_Minimum_Hours_of_Training_to_Win_a_Competition/minimum_hours_of_training_to_win_a_competition.cpp
class Solution {
public:
    int minNumberOfHours(int initialEnergy, int initialExperience, vector<int>& energy, vector<int>& experience) {
        
        // parameters
        int EnergySum = 0;
        int ExtraExp = 0, ExpGap;

        // start computing
        for (int id=0; id<energy.size(); id++) {

            // compute energy
            EnergySum += energy[id];

            // compute experience
            if (experience[id] >= initialExperience) {

                ExpGap = experience[id] - initialExperience + 1;
                ExtraExp += ExpGap;
                initialExperience += experience[id] + ExpGap;

            }
            else initialExperience += experience[id];

        }

        return (std::max((EnergySum - initialEnergy + 1), 0) + ExtraExp);

    }
};
